core/src/main/scala/dagr/core/config/Configuration.scala

```diff
@@ -166,24 +166,30 @@ private[config] trait ConfigurationLike extends LazyLogging {
   }
 
   /**
-    * Attempts to determine the path to an executable, first by looking it up in config,
-    * and if that fails, by attempting to locate it on the system path. If both fail then
-    * an exception is raised.
+    * Attempts to determine the path to an executable, first by looking it up in the config and optionally checking to
+    * see if the path exists. If that fails, we will then locate the executable on the system path. If both strategies
+    * fail, then an exception is raised.
     *
     * @param path the configuration path to look up
     * @param executable the default name of the executable
-    * @return An absolute path to the executable to use
+    * @param mustExist ensure the configuration path to look up references a file which exists
```

Comment:
```suggestion
    * @param allowMissingConfigReference true to require that the executable path in the configuration must exist, false fallback to the system path if the configuration path exists but not the returned executable path.
```
I don't _really_ like longer names, but in this case I asked "what must exist?" prior to reading the docs, so what do think about a longer name?  Also, I tried to be more verbose about the behavior, but feel free to wordsmith further.